ui/ozone/platform/wayland/test/mock_surface.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef UI_OZONE_PLATFORM_WAYLAND_TEST_MOCK_SURFACE_H_
+#define UI_OZONE_PLATFORM_WAYLAND_TEST_MOCK_SURFACE_H_
+
+#include <wayland-server-protocol-core.h>
+
+#include "base/macros.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "ui/ozone/platform/wayland/test/mock_xdg_surface.h"
+#include "ui/ozone/platform/wayland/test/server_object.h"
+
+struct wl_resource;
+
+namespace wl {
+
+extern const struct wl_surface_interface kMockSurfaceImpl;
+
+// Manage client surface
+class MockSurface : public ServerObject {
+ public:
+  explicit MockSurface(wl_resource* resource);
+  ~MockSurface() override;
+
+  static MockSurface* FromResource(wl_resource* resource);
+
+  MOCK_METHOD3(Attach, void(wl_resource* buffer, int32_t x, int32_t y));
+  MOCK_METHOD4(Damage,
+               void(int32_t x, int32_t y, int32_t width, int32_t height));
+  MOCK_METHOD0(Commit, void());
+
+  void set_xdg_surface(std::unique_ptr<MockXdgSurface> xdg_surface) {
+    xdg_surface_ = std::move(xdg_surface);
+  }
+  MockXdgSurface* xdg_surface() { return xdg_surface_.get(); }
```

Comment:
() const?

Reply:
Could be.  Will review all headers and add const where appropriate.